### ISSUE TITLE
v1.0.0: links not working after initial load

### ISSUE THREAD

**Issue by dragonfly4** [davidjbradshaw/image-map-resizer#34]
Tested in Chrome and Safari. I received a complaint/notice that the homepage links were not clickable when returning to the homepage so I started extensive testing myself.

1) go to homepage that has imagemap, all links work fine.
2) click on a link
3) return to homepage by clicking on home or top logo --> imagemap links are not clickable anymore. No error in Chrome dev console.

I have no idea what the reason is so reverted back to v0.6.1 and I cannot simulate it anymore for the moment. Will let you know if I would see similar behaviour with v0.6.1

**Comment by griogon** [comment]
I believe this is the same issue as https://github.com/davidjbradshaw/image-map-resizer/issues/33

Happens in Firefox, too. I am able to reproduce reliably by pressing Ctrl-Shift-R (force reload).

imageMapResizer is setting the coord values to "Infinity" -- e.g., changing the following original html:
<area shape="poly" coords="73,21,72,77,118,76,118,24"></area>

into this:
<area shape="poly" coords="Infinity,Infinity,Infinity,Infinity,Infinity,Infinity,Infinity,Infinity"></area>

If the user then minimizes the window, then brings it back up, the coords are corrected and everything works as it should. (There are other actions which have the same result, such as clicking on the location bar, then clicking back onto the page.)

I was able to determine that the coords were being changed to "Infinity" using Firebug. To see for yourself:
1) open firebug, and inspect one of the <area> elements (at this point the coords should look fine, because opening Firebug causes the coords to get fixed, just like minimizing and unminimizing the window)
2) press Ctrl-Shift-R
3) Firebug should refresh but still have the same <area> selected ... you should notice at this point that the coords are set to "Infinity" values.

**Comment by roumenkey** [comment]
I've tried to use this script today only to experience the same problem. The links will not show up on the initial load and change to 
`<area shape="rect" coords="Infinity,NaN,Infinity,NaN" href="http://www.address.com> `
instead. If I resize the browser window or switch tabs back and forth those mapped links will immediately appear. Tried Chrome, Safari, Firefox. Unfortunately the other script from Matt Stove isn't working for me at all so I'm stuck for now. I bet it's just one stupid setting and it will work. Is anybody here willing to help and fix this script? It would be greatly appreciated.

**Comment by dragonfly4** [comment]
I'm still using v0.6.1
https://github.com/davidjbradshaw/image-map-resizer/releases/tag/v0.6.1

**Comment by roumenkey** [comment]
I've tried that too. same result.

**Comment by dragonfly4** [comment]
In that case please create a new issue, this is related to v1.0 :)

V0.6.1 is working fine for me

**Comment by roumenkey** [comment]
actually, I've tried this version from a different source and that didn't work. now I've tried it again using the link dragonfly4 provided and it works!

thank you dragonfly4!

**Comment by MicahBrown** [comment]
I had this issue too. I had to add another event listener `image.addEventListener('load',  resizeMap, false);` to [this function](https://github.com/davidjbradshaw/image-map-resizer/blob/master/js/imageMapResizer.js#L48).

**Comment by davidjbradshaw** [comment]
@MicahBrown do you want to make a PR for that?

**Comment by MicahBrown** [comment]
[Made one](https://github.com/davidjbradshaw/image-map-resizer/pull/36). Is that all you need?

**Comment by davidjbradshaw** [comment]
I've just added the follow which will hopefully fix this.

```js
window.addEventListener('readystatechange', resizeMap,  false);
```

So now if the page isn't ready when the code first runs, it will detect every change to read state and update itself.

Released as v1.0.1 - Please let me know if you still have problems with this.

**Comment by davidjbradshaw** [comment]
Just noticed another bug, the image load listerner was listerning for `onload` rather than `load`.

So v1.0.2 and hopefully that is every base now covered.
